tests/cases/test_propagate.rs

```diff
@@ -0,0 +1,81 @@
+use common::*;
+use parameterized_macro::parameterized;
+use std::path::PathBuf;
+
+fn publish_components(state: &TestState, env_name: &str) -> lal::LalResult<PathBuf> {
```

Comment:
Surprisingly, rust is okay with this function name shadowing `common::publish_components`.

I wonder if it is because
a) this function doesn't call `common::publish_components`, and so there's no confusion to mistake it for recursion
b) rust can distinguish between function arity? (the state of https://github.com/rust-lang/rfcs/issues/1586 suggests not)
c) we can still reference the other function with the `common::` prefix aka. universal function call syntax.

The `use *` syntax is weird.